libnetwork/drivers/bridge/bridge_linux.go

```diff
@@ -410,6 +406,55 @@ func (n *bridgeNetwork) iptablesEnabled(version iptables.IPVersion) (bool, error
 	return n.driver.config.EnableIPTables, nil
 }
 
+func (n *bridgeNetwork) newIptablesNetwork() error {
+	config4, err := makeNetworkConfigFam(n.config.HostIPv4, n.bridge.bridgeIPv4, n.gwMode(iptables.IPv4))
+	if err != nil {
+		return err
+	}
+	config6, err := makeNetworkConfigFam(n.config.HostIPv6, n.bridge.bridgeIPv6, n.gwMode(iptables.IPv6))
+	if err != nil {
+		return err
+	}
+	iptn, err := newIptablesNetwork(networkConfig{
+		IfName:     n.config.BridgeName,
+		Internal:   n.config.Internal,
+		ICC:        n.config.EnableICC,
+		Masquerade: n.config.EnableIPMasquerade,
+		Config4:    config4,
+		Config6:    config6,
+		Hairpin:    !n.driver.config.EnableUserlandProxy || n.driver.config.UserlandProxyPath == "",
+		Enable4:    n.driver.config.EnableIPTables,
+		Enable6:    n.driver.config.EnableIP6Tables,
+	})
+	if err != nil {
+		return err
+	}
+	n.iptablesNetwork = iptn
```

Comment:
If you add an extra `Reload` method, then you don't need to re-assign `n.iptablesNetwork` in the firewalld reload code, and you can make this simpler:

```suggestion
    return newIptablesNetwork(networkConfig{
		IfName:     n.config.BridgeName,
		Internal:   n.config.Internal,
		ICC:        n.config.EnableICC,
		Masquerade: n.config.EnableIPMasquerade,
		Config4:    config4,
		Config6:    config6,
		Hairpin:    !n.driver.config.EnableUserlandProxy || n.driver.config.UserlandProxyPath == "",
		Enable4:    n.driver.config.EnableIPTables,
		Enable6:    n.driver.config.EnableIP6Tables,
	})
```

And let the caller assign `n.iptablesNetwork`.

Reply:
I think this is reiterating your [earlier comment](https://github.com/moby/moby/pull/49668#discussion_r2015960906), and discussion of [a possible Reload](https://github.com/moby/moby/pull/49668#discussion_r2012825481) method  ...

This function's really part of the bridgeNetwork's constructor. Moving the assignment out means the driver ends up calling the bridgeNetwork to construct the object - then making the assignment to a bridgeNetwork property itself.

So, your suggested simplification just moves the code to the caller, and `driver` needs to be involved in the innards of `bridgeNetwork` ...

```
        bridgeSetup.queueStep(func(*networkConfiguration, *bridgeInterface) error {
-               return network.newIptablesNetwork()
+               n, err := network.newIptablesNetwork()
+               if err != nil {
+                       return err
+               }
+               network.iptablesNetwork = n
+               return nil
        })
```

(So, I don't like it! But, it's not a big deal - I'll rearrange it anyway to get this unblocked. Can always rearrange it back later, it's all internal to the bridge driver.)

Reply:
Done.